### PR TITLE
Redirect to originally requested page after logging in 

### LINKED ISSUE #343: Redirect to originally requested page after logging in
Current always goes to home

### PR DESCRIPTION
**Pull request by navandstokes**
Closes #343

### CODE REVIEW

File: app/frontend/src/features/auth/login/Login.tsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ export default function Login() {
   const error = useTypedSelector((state) => state.auth.error);
 
   const location = useLocation();
+  const { from } = (location.state as any) || { from: { pathname: "/" } };
```
Comment (amdwit) [inline app/frontend/src/features/auth/login/Login.tsx]: I was thinking about going straight to the needed value and not creating an object.
```suggestion
  const location = useLocation<undefined | { from: Location }>();
  const redirectTo = location.state?.from.pathname || "/";
```

And then also change the redirect.to

Comment (navandstokes) [reply app/frontend/src/features/auth/login/Login.tsx]: Trying to understand what is happening here. `state?.` tells typescript to stop checking the rest. This `<undefined | { from: Location }>` means the value returned can be undefined or an object. But can you explain the `{ from: Location }` part?

Comment (amdwit) [reply app/frontend/src/features/auth/login/Login.tsx]: `useLocation<undefined | { from: Location }>` means that the `state` property of the returned location is either undefined or an object with a `from` property containing a value of type `Location`.

**Review by amdwit — APPROVED** [review]

**Review by lucaslcode — APPROVED** [review]
Feel free to merge (or do you want me to?)

**Pull request merged**